File: packages/contracts/python/noa_contracts/critique_report.py
from __future__ import annotations

from typing import Literal
from pydantic import BaseModel


class CritiqueIssue(BaseModel):
    code: str
    message: str
    severity: str
    element_id: str | None = None
    model_config = {"extra": "allow"}


class CritiqueScores(BaseModel):
    plan_fidelity: float
    inspiration_alignment: float
    architectural_plausibility: float
    geometric_consistency: float
    render_coherence: float


class CritiqueReport(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    canonical_model_id: str
    critique_job_id: str
    scores: CritiqueScores
    issues: list[CritiqueIssue]
    suggestions: list[str]
    blocking: bool
    created_at: str

File: packages/contracts/v1/critique-report.ts
import type { UUID, ISO8601 } from "./common.js";

export interface CritiqueIssue {
  code: string;
  message: string;
  severity: string;
  element_id?: string;
  [key: string]: unknown;
}

export interface CritiqueReport {
  schema_version: "1.0";
  id: UUID;
  canonical_model_id: UUID;
  critique_job_id: UUID;
  scores: {
    plan_fidelity: number;
    inspiration_alignment: number;
    architectural_plausibility: number;
    geometric_consistency: number;
    render_coherence: number;
  };
  issues: CritiqueIssue[];
  suggestions: string[];
  blocking: boolean;
  created_at: ISO8601;
}
